### PR TITLE
format request the raven way. close #17

### LINKED ISSUE #17: format request "the raven way"
Just a suggestions:
I think Sentry displays a nicer UI if you format the request sent to sentry the right way (https://github.com/getsentry/raven-node/blob/master/lib/middleware/connect.js#L21)

maybe you can simply pass `request.raw.req` to `node-raven`, to get that data. 

The parser logic for it can be found here: https://github.com/getsentry/raven-node/blob/8c7303c2d1cdfe087042c80a3b74d638ca352445/lib/parsers.js#L66
## request right now

image
## in the right format

image

### PR DESCRIPTION
**Pull request by fiws**
basic implementation of sentry http interface.

### CODE REVIEW

**Comment by fiws** [comment]
ah tests need to be adjusted

**Comment by fiws** [comment]
i just noticed that `request.info.uri` is not always set. closing for now.

reopening when this is resolved.

File: src/index.js [hunk 1]
```diff
@@ -6,15 +6,23 @@ exports.register = function (server, options, next) {
   var client = new raven.Client(options.dsn, options.client);
   server.expose('client', client);
   server.on('request-error', function (request, err) {
+    var baseUrl = request.info.uri;
```
Comment (bendrucker) [inline src/index.js]: For parity with the native/express parser please don't use this value unless it's available 100% of the time:

https://github.com/getsentry/raven-node/blob/master/lib/parsers.js#L110

File: src/index.js [hunk 2]
```diff
@@ -6,15 +6,23 @@ exports.register = function (server, options, next) {
...
+    if (baseUrl === undefined) {
+      if (request.info.host) baseUrl = server.info.protocol + '://' + request.info.host;
+      else baseUrl = server.info.uri;
+    }
     (client.captureException || client.captureError).call(client, err, {
+      request: {
```
Comment (bendrucker) [inline src/index.js]: Please sort these to match the Raven parser for easy comparison:

https://github.com/getsentry/raven-node/blob/master/lib/parsers.js#L154-L162

File: src/index.js (hunk 1 above)
Comment (fiws) [inline src/index.js]: it's not available in some cases, but that's what L10-13 is for oO
info.uri is configurable when setting up hapi, and falls back to browser sent stuff.

**Comment by fiws** [comment]
rebased, sorry for the delay. Are the tests the only thing that is missing now?

File: index.js [hunk 3]
```diff
@@ -6,15 +6,25 @@ exports.register = function (server, options, next) {
   var client = new raven.Client(options.dsn, options.client)
   server.expose('client', client)
   server.on('request-error', function (request, err) {
+
+    var baseUrl = request.info.uri;
+    if (baseUrl === undefined) {
```
Comment (bendrucker) [inline index.js]: `baseUrl = baseUrl || request.info.host && ... || server.info.uri`

Comment (fiws) [inline index.js]: ok, i think it's kind of complicated looking now, but your choice.

**Pull request closed without merging**